docs/design/dfx-extensions.md

```diff
@@ -0,0 +1,66 @@
+
+2. Download and Extraction:
+    - Once the compatible version of the extension is determined, dfx constructs a download URL. This URL points to a GitHub releases page where the extensions are hosted. Currently, extensions support downloading only from `dfinity/dfx-extensions` repository. Here is how the URL template looks like:
+        ```
+        "https://github.com/dfinity/dfx-extensions/releases/download/{github_release_tag}/{extension_archive_name}.tar.gz";
+        ```
+        
+    - The extension is then downloaded from this URL. If the download fails for any reason, an error is returned.
+    - After successful download, the extension, which is in a compressed archive format (.tar.gz), is unpacked to a temporary directory.
+
+3. Final Installation:
+    - Post-extraction, the extension is renamed and moved to its permanent location (the directory where extensions are meant to reside).
+    - If the user specifies a custom name for the extension at the time of installation, the binary is renamed to reflect this custom name.
+    - On UNIX systems, appropriate permissions are set for the installed extension to ensure it's executable.
+
+4. Error Handling:
+    - Throughout this process, any potential errors are diligently checked for, ensuring a robust installation process. Examples include errors related to incompatible versions, download failures, decompression errors, etc.
+    - This process ensures that the extensions are not only installed from a trusted source but are also compatible with the user's dfx version, enhancing the utility's reliability and user experience.
+
+## Running Extensions
+
+Installed extensions are stored in `dfx`'s cache. Let's say you have `sns` extension installed; here is how the cache directory structure will look like:
+```console
+❯ tree $(dfx cache show)
+~/.cache/dfinity/versions/0.15.0
+└── extensions
+   └── sns
+      ├── CHANGELOG.md
+      ├── extension.json
+      ├── LICENSE
+      ├── README.md
+      ├── sns
+      └── sns-cli
+```
+
+When requested to run an extension, dfx fetches the location of the executable associated with the extension, by determining the cache location associated with its current version, and searching for directory with extension name inside: `$(dfx cache show)/extensions/EXT_NAME/EXT_NAME`. Fox example , when you're running `dfx sns ARGS` or `dfx extension run sns ARGS`, `dfx` will try to find the extension binary at this path: `$(dfx cache show)/extensions/sns/sns`. 
+
+When executing the binary, the path to the cache is appended as an argument when invoking the extension's executable, in practice that means the extension will always be executed with `--dfx-cache-path $(dfx cache show)` parameter, like so: `exec $(dfx cache show)/extensions/sns/sns --dfx-cache-path $(dfx cache show)`. This allows the extension to depend on other extensions, and binaries stored in dfx's cache.
```

Comment:
Rather than prose, it would be better to describe which parameters dfx passes to the extension binary in the form of a table or list.  A prose description / example can follow.

Otherwise, buried in the middle of the paragraph is the fact that dfx passes `--dfx-cache-path "$(dfx cache show)"`.